src/utils/packetIdAllocator.ts

```diff
@@ -0,0 +1,79 @@
+/* Keeps track of free and used packet ids by storing available IDs as intervals */
```

Comment:
nit: IDs vs ids. I favor IDs

Reply:
also it is not clear what an "interval" is. Can you add better documentation? 

At first glance this looks like a way to make sure that we have unique IDs for Messages. I like the initiative on creating this but do we have performance metrics on it? Is it better than something out there that is already implemented that accomplishes the same goal?